src/overlay005/field_control.c

```diff
@@ -0,0 +1,1092 @@
+static BOOL Field_MapConnection(const FieldSystem *fieldSystem, int playerX, int playerZ, Location *nextMap);
+static void Field_TrySetMapConnection(FieldSystem *fieldSystem);
+static BOOL Field_DistortionInteract(FieldSystem *fieldSystem, MapObject **object);
+static int Field_CheckTrainerInfo(void);
+
+static void FieldInput_Clear (FieldInput *input)
+{
+    input->interact = FALSE;
+    input->endMovement = FALSE;
+    input->menu = FALSE;
+    input->registeredItem = FALSE;
+    input->sign = FALSE;
+    input->mapTransition = FALSE;
+    input->movement = FALSE;
+    input->unused1 = FALSE;
+    input->unused2 = FALSE;
+    input->unused3 = FALSE;
+    input->unused4 = FALSE;
+    input->unused5 = FALSE;
+    input->playerDir = DIR_NONE;
+    input->transitionDir = DIR_NONE;
+}
+
+void FieldInput_Update (FieldInput *input, FieldSystem *fieldSystem, u16 pressedKeys, u16 heldKeys)
+{
+    FieldInput_Clear(input);
+
+    int moveState = Player_MoveState(fieldSystem->playerAvatar);
+    int avatarState = PlayerAvatar_MoveState(fieldSystem->playerAvatar);
+    int playerDir = PlayerAvatar_GetDir(fieldSystem->playerAvatar);
+
+    input->pressedKeys = pressedKeys;
+    input->heldKeys = heldKeys;
+
+    if (moveState == PLAYER_MOVE_STATE_END || moveState == PLAYER_MOVE_STATE_NONE) {
+        if (pressedKeys & PAD_BUTTON_X) {
+            input->menu = TRUE;
+        }
+
+        if (pressedKeys & PAD_BUTTON_Y) {
+            input->registeredItem = TRUE;
+        }
+
+        if (pressedKeys & PAD_BUTTON_A) {
+            input->interact = TRUE;
+        }
+
+        if (pressedKeys & PAD_BUTTON_B) {
+            input->unused1 = TRUE;
+        }
+
+        if (heldKeys & (PAD_KEY_UP | PAD_KEY_DOWN | PAD_KEY_LEFT | PAD_KEY_RIGHT)) {
+            input->sign = TRUE;
+        }
+
+        if (heldKeys & (PAD_KEY_UP | PAD_KEY_DOWN | PAD_KEY_LEFT | PAD_KEY_RIGHT)) {
+            input->mapTransition = TRUE;
+        }
+    }
+
+    if (moveState == PLAYER_MOVE_STATE_END && avatarState == AVATAR_MOVE_STATE_MOVING) {
+        input->movement = TRUE;
+    }
+
+    if (moveState == PLAYER_MOVE_STATE_END) {
+        input->endMovement = TRUE;
+    }
+
+    if (playerDir == DIR_NORTH && heldKeys & PAD_KEY_UP || 
+        playerDir == DIR_SOUTH && heldKeys & PAD_KEY_DOWN || 
+        playerDir == DIR_WEST  && heldKeys & PAD_KEY_LEFT || 
+        playerDir == DIR_EAST  && heldKeys & PAD_KEY_RIGHT) {
+
+        input->transitionDir = playerDir;
+    } else {
+        input->transitionDir = DIR_NONE;
+    }
+
+    input->playerDir = sub_02061308(fieldSystem->playerAvatar, pressedKeys, heldKeys);
+}
+
+int FieldInput_Process (const FieldInput *input, FieldSystem *fieldSystem)
+{
+    if (input->unused5 == FALSE && sub_0203F5C0(fieldSystem, 1) == TRUE) {
+        return TRUE;
+    }
+
+    if (input->unused5 == FALSE) {
+        BOOL doublesCheck = sub_02054AB0(Party_GetFromSavedata(fieldSystem->saveData));
```

Comment:
Why name this `doublesCheck`?

Reply:
At minimum, give the function a meaningful name and the returned value here a name that makes sense semantically in English. (e.g. `if (playerHasTwoMons)` or some more brief description)